### PR TITLE
wallet2: store testnet bool in keys file

### PR DESCRIPTION
**Pull request by Jaqueeee**

### CODE REVIEW

**Comment by Jaqueeee** [comment]
OK. I agree in general. Started working on the genesis way of determining testnet or mainet, but realized it wouldn't work in those cases wallet cache is deleted and wallet doesn't have a genesis block. But maybe this could be seen as a corner case where the user is required to set testnet flag manually? Otherwise I guess it could be determined by parsing the wallet address somehow.
@moneromooo-monero

**Comment by moneromooo-monero** [comment]
That's a good point... OK, then make sure that at load time you check for inconsistency, as well as when changing it or genesis (if applicable).

**Comment by moneromooo-monero** [comment]
Or maybe store the genesis block in the keys instead ?

**Comment by Jaqueeee** [comment]
Added a check in load_keys. 

TL:DR: This patch prevents wallets without cache files to be opened/converted to wrong wallet format (testnet/mainnet). And also make it possible to open a testnet wallet without a --testnet flag. 

new wallet = wallet created after this PR is merged

**Old wallets with cache file**
mainnet wallet with --testnet flag - **FAIL:** Genesis mismatch
testnet wallet without --testnet flag - **FAIL:** Genesis mismatch
**Old wallets without cache file**
mainnet wallet with --testnet flag - **SUCCESS:** wallet will be converted to testnet wallet **(!)**
testnet wallet without --testnet flag - **SUCCESS:** wallet will be converted to mainnet wallet **(!)**

**New wallets with cache file**
mainnet wallet with --testnet flag - **FAIL**: testnet flag != testnet bool in .keys
testnet wallet without --testnet flag - **SUCESS** : wallet will be opened as a testnet wallet
**New wallets without cache file**
mainnet wallet with --testnet flag - **FAIL:** testnet flag != testnet bool in .keys
testnet wallet without --testnet flag - **SUCCESS** : wallet will be opened as a testnet wallet

**Comment by moneromooo-monero** [comment]
If you run without --testnet flag, and load a testnet wallet, does it auto change all the stuff to testnet ? That is, change daemon port to 28081, start a testnet daemon, etc, etc ? That seems a fair bit more complicated and unintuitive than erroring out, no ?

**Comment by Jaqueeee** [comment]
GUI would do all that. And also show the "TESTNET" label in red, next to the balance. I'm not sure if cli wallet does anything like that. 

But maybe it's better to just require the --testnet flag. I can change the automatic switching to an error instead.

**Comment by moneromooo-monero** [comment]
It does sound useful not to have a state, but it needs to be sure everything works as expected, and there seems to be a number of things that are dependent on the testnet/not flag. I just feel wary of this, but apart from the above, I do not have precise objections.

**Comment by Jaqueeee** [comment]
useful indeed. But that logic can wait. Changed to only store the bool and error out on loading failure as suggested.

File: src/wallet/wallet2.cpp [hunk 1]
```diff
@@ -1940,6 +1940,9 @@ bool wallet2::store_keys(const std::string& keys_file_name, const std::string& p
   value2.SetInt(m_merge_destinations ? 1 :0);
   json.AddMember("merge_destinations", value2, json.GetAllocator());
 
+  value2.SetInt(m_testnet ? 1 :0);
```
Comment (vtnerd) [inline src/wallet/wallet2.cpp]: rapidjson has a `SetBool` / `GetBool` - that seems preferable ... ?

Comment (moneromooo-monero) [inline src/wallet/wallet2.cpp]: Interesting. I remember looking, and not finding one... It did seem wtf, so maybe I should have looked twice :)
